<h1>Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: John Kevin Aguilera</h3>

In [1]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p> Importing datasets to Jupyter notebook</p>
<p> Converting datasets to dataframe </p>
<ul>Requirements
    <li>Name of the dataframe</li>
    <li>Source</li>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>How to fillup Null values</h1>

In [4]:
chinstrap_weight = penguins[(penguins['species'] == "Chinstrap") & (penguins['body_mass_g'] >= 4000)]

<h5>Filling up bill_lenght_mm</h5>

In [5]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [6]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [7]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [8]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<p>Filling up sex</p>

In [9]:
penguins['sex'].mode()

0    male
Name: sex, dtype: object

In [10]:
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [11]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [12]:
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [13]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [14]:
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [15]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [16]:
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [17]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>Descriptive Analysis</h1> 

In [18]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.443643       1.969027          14.020657   799.613058   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [19]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     344.0    43.921930    5.443643    32.1    39.275    44.25   
bill_depth_mm      344.0    17.151170    1.969027    13.1    15.600    17.30   
flipper_length_mm  344.0   200.915205   14.020657   172.0   190.000   197.00   
body_mass_g        344.0  4201.754386  799.613058  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the dataframe</h3>

In [20]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h2>Selecting the column</h2>
<h3>Syntax</h3><br>
data_frame["column_name"]

<h1>Creating dataframes</h1>

In [21]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [22]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [23]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen        39.10000       18.70000         181.000000   
1    Adelie  Torgersen        39.50000       17.40000         186.000000   
2    Adelie  Torgersen        40.30000       18.00000         195.000000   
3    Adelie  Torgersen        43.92193       17.15117         200.915205   
4    Adelie  Torgersen        36.70000       19.30000         193.000000   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream        36.60000       18.40000         184.000000   
148  Adelie      Dream        36.00000       17.80000         195.000000   
149  Adelie      Dream        37.80000       18.10000         193.000000   
150  Adelie      Dream        36.00000       17.10000         187.000000   
151  Adelie      Dream        41.50000       18.50000         201.000000   

     body_mass_g     sex  year  
0    3750.000000    male  2007  
1    3800.000000  female  2007  
2    3250.000000  female  2007  
3    4201.754386    male  2007  
4    3450.000000  female  2007  
..           ...     ...   ...  
147  3475.000000  female  2009  
148  3450.000000  female  2009  
149  3750.000000    male  2009  
150  3700.000000  female  2009  
151  4000.000000    male  2009  

[152 rows x 8 columns]

In [24]:
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [25]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  
152  4500.000000  female  2007  
153  5700.000000    male  2007  
154  4450.000000  female  2007  
155  5700.000000    male  2007  
156  5400.000000    male  2007  
..           ...     ...   ...  
271  4201.754386    male  2009  
272  4850.000000  female  2009  
273  5750.000000    male  2009  
274  5200.000000  female  2009  
275  5400.000000    male  2009  

[124 rows x 8 columns]

In [26]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [27]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [28]:
penguins['body_mass_g'] >= 4000

0      False
1      False
2      False
3       True
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [29]:
chinstrap_weight.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>How to change data type of a column</h1>

In [30]:
penguins['species'] = penguins['species'].astype('category')
penguins['year'] = penguins['year'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [31]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [32]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

In [33]:
penguins['mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [34]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  mass_kg  
0    3750.000000    male  2007      3.8  
1    3800.000000  female  2007      3.8  
2    3250.000000  female  2007      3.2  
3    4201.754386    male  2007      4.2  
4    3450.000000  female  2007      3.4  
..           ...     ...   ...      ...  
339  4000.000000    male  2009      4.0  
340  3400.000000  female  2009      3.4  
341  3775.000000    male  2009      3.8  
342  4100.000000    male  2009      4.1  
343  3775.000000  female  2009      3.8  

[344 rows x 9 columns]

<h1>Exploratory Data Analytics (EDA)</h1>

<h1>1. Which island has the most penguins?</h1>

In [35]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [36]:
penguins.groupby(['island']).size()

island
Biscoe       168
Dream        124
Torgersen     52
dtype: int64

<h3>Insight #1: The island that has the most penguins recorded is <strong>Biscoe.</h3>

<h1>2. Which year and species recorded the highest number of penguins with over 3500 body mass.</h1>

In [37]:
penguins['year'].unique()

[2007, 2008, 2009]
Categories (3, int64): [2007, 2008, 2009]

In [38]:
Over3500 = penguins[penguins['body_mass_g'] >= 3500]

In [39]:
bmass_peryear = Over3500.groupby(['species','year']).size()

In [40]:
bmass_peryear

species    year
Adelie     2007    30
           2008    38
           2009    30
Chinstrap  2007    20
           2008    14
           2009    17
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

<h3>Insight #2: <b>Gentoo</b>  has the highest number of penguins with over 3500 body mass on <b>2008</n>.</h3>

<h1>3. Which species has the highest average bill length, bill depth, body mass?</h1>

In [41]:
penguins['species'].unique()

['Adelie', 'Gentoo', 'Chinstrap']
Categories (3, object): ['Adelie', 'Chinstrap', 'Gentoo']

In [42]:
BLength_PerSpecies = penguins.groupby(['species',])['bill_length_mm'].mean()

In [43]:
BLength_PerSpecies

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

In [44]:
Bdepth_PerSpecies = penguins.groupby(['species'])['bill_depth_mm'].mean()

In [45]:
Bdepth_PerSpecies

species
Adelie       18.338495
Chinstrap    18.420588
Gentoo       14.999606
Name: bill_depth_mm, dtype: float64

In [46]:
Bmass_PerSpecies = penguins.groupby(['species'])['body_mass_g'].mean()

In [47]:
Bmass_PerSpecies

species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h3>Insight #3: <b>Chinstrap</b> had the longest bill length and deepest bill depth, while <b>Gentoo</b> had the heaviest body mass.</h3>

<h1>Question #4: What is the demography of Torgersen Island, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [48]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [49]:
torgersen_df = penguins[penguins['island'] == 'Torgersen']

In [50]:
torgersen_df.groupby(['species']).size()

species
Adelie       52
Chinstrap     0
Gentoo        0
dtype: int64

In [51]:
torgersen_df.groupby(['species','year']).size()

species    year
Adelie     2007    20
           2008    16
           2009    16
Chinstrap  2007     0
           2008     0
           2009     0
Gentoo     2007     0
           2008     0
           2009     0
dtype: int64

<h3>Insight #4: The island <b>Torgersen</b> only had <b>Adelie</b> as its inhabitant, with <b>2007</b> being the highest number of Adelies recorded.</h3>

<h1>Question #5: What is the demography of Island Biscoe, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [52]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [53]:
biscoe_df = penguins[penguins['island'] == 'Biscoe']

In [54]:
biscoe_df.groupby(['species']).size() # the number of species inhabiting the island.

species
Adelie        44
Chinstrap      0
Gentoo       124
dtype: int64

In [55]:
biscoe_df.groupby(['species','year']).size()

species    year
Adelie     2007    10
           2008    18
           2009    16
Chinstrap  2007     0
           2008     0
           2009     0
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

<h3>Insight #5: The island <b>Biscoe</b> had <i>44 Adelie</i> and <i>124 Gentoo</i> as its inhabitants, with the year <b>2008</b> recording <i>18 Adelies</i> and <b>2009</b> recording <i>44 Gentoos</i>.</h3>

<h1>Question #6: What is the demography of Island Dream, and which year had the highest number of Adelie, Gentoo, and Chinstrap species recorded? </h1>

In [56]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [57]:
dream_df = penguins[penguins['island'] == 'Dream']

In [58]:
dream_df.groupby(['species']).size()

species
Adelie       56
Chinstrap    68
Gentoo        0
dtype: int64

In [59]:
dream_df.groupby(['species','year']).size()

species    year
Adelie     2007    20
           2008    16
           2009    20
Chinstrap  2007    26
           2008    18
           2009    24
Gentoo     2007     0
           2008     0
           2009     0
dtype: int64

<h3>Insight #6: The island <b>Dream</b> has <i>56 Adelie</i> and <i>68 Chinstrap</i> penguins as its inhabitants, with the year <b>2007</b> recording the highest number of Adelie and Chinstrap penguins.</h3>

<h1>Question #7: Which species has the highest number of male and female?</h1>

In [60]:
adelie_df['sex'].value_counts()

sex
male      79
female    73
Name: count, dtype: int64

In [61]:
gentoo_df['sex'].value_counts()

sex
male      66
female    58
Name: count, dtype: int64

In [62]:
chinstrap_df['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

<h3>Insight #7: The species <b>Adelie</b> recorded the highest number of male and female with 79 and 73 respectively.</h3>

<h1>8. What is the average body mass of each penguins the lives on each island?</h1>

In [63]:
penguins['island'].unique()

['Torgersen', 'Biscoe', 'Dream']
Categories (3, object): ['Biscoe', 'Dream', 'Torgersen']

In [64]:
Bmass_PerIsland = penguins.groupby(['island','species'])['body_mass_g'].mean()

In [65]:
Bmass_PerIsland

island     species  
Biscoe     Adelie       3709.659091
           Chinstrap            NaN
           Gentoo       5068.965761
Dream      Adelie       3688.392857
           Chinstrap    3733.088235
           Gentoo               NaN
Torgersen  Adelie       3715.899123
           Chinstrap            NaN
           Gentoo               NaN
Name: body_mass_g, dtype: float64

<h3>Insigth #8: The average body mass of each penguins that live on each island are as follows:</h3>
<p>Biscoe: Adelie has an average body mass of <b> 3709.659091</b> while Gentoo has an average body mass of 5068.965761</p>
<p>Dream: Adelie has an average body mass of <b>3688.392857</b> while Chinstrap has an average body mass of <b>3733.088235
</b></p>
<p>Torgersen: Adelie has an average body mass of <b> 3715.899123</p>

<h1>Question #9: We found out earlier that Chinstrap is the species that has the highest average bill length and depth while Gentoo has the highest average body mass. Does the bill length and depth of Chinstrap affects it's body mass? Does Gentoo having heaviest average body mass affect the growth of it's limbs?</h1>

<h5>Finding correlation on Chinstrap</h5>

In [66]:
ch_blength_bmass_corr = chinstrap_df['bill_length_mm'].corr(chinstrap_df['body_mass_g'])

In [67]:
ch_blength_bmass_corr

np.float64(0.5136383479489105)

In [68]:
ch_bdepth_bmass_corr =  chinstrap_df['bill_depth_mm'].corr(chinstrap_df['body_mass_g'])

In [69]:
ch_bdepth_bmass_corr 

np.float64(0.6044982582238205)

<h5>Finding correlation on Gentoo</h5>

In [70]:
ge_blength_bmass_corr = gentoo_df['bill_length_mm'].corr(gentoo_df['body_mass_g'])

In [71]:
ge_blength_bmass_corr

np.float64(0.6736363731181065)

In [72]:
ge_bdepth_bmass_corr = gentoo_df['bill_depth_mm'].corr(gentoo_df['body_mass_g'])

In [73]:
ge_bdepth_bmass_corr

np.float64(0.6665426960319999)

<h3>Insigth #9: According to the data above, there is a correlation between the body mass and bill length and depth of both <b> Gentoo</b> and <b>Chinstrap</b></h3>

<h1>Question #10: Which island has the lowest average bill length, bill depth, and body mass?</h1>

In [96]:
BLength_PerIsland = penguins.groupby(['island'])['bill_length_mm'].mean()

In [97]:
BLength_PerIsland

island
Biscoe       45.249535
Dream        44.167742
Torgersen    39.046576
Name: bill_length_mm, dtype: float64

In [99]:
Bdepth_PerIsland = penguins.groupby(['island'])['bill_depth_mm'].mean()

In [100]:
Bdepth_PerIsland

island
Biscoe       15.882447
Dream        18.344355
Torgersen    18.404830
Name: bill_depth_mm, dtype: float64

In [102]:
Bmass_PerIsland = penguins.groupby(['island'])['body_mass_g'].mean()

In [103]:
Bmass_PerIsland

island
Biscoe       4712.956871
Dream        3712.903226
Torgersen    3715.899123
Name: body_mass_g, dtype: float64

<h3>Insigth #10: According to the data below, we found out that:</h3>
<p> <b>Torgersen</b> has the lowest average biil length, <b>Biscoe</b> has the lowest average bill depth, and <b>Dream</b> has the lowest average body mass</p>

<h1>Question #11: Which year recorded the highest number of female per island?</h1>

In [119]:
female_PerYear = penguins.groupby(['sex','year']).size()

In [123]:
female_PerYear 

sex     year
female  2007    51
        2008    56
        2009    58
male    2007    59
        2008    58
        2009    62
dtype: int64

<h3>Insigth #11: According to the data above, we found out that:</h3>
<p>The year <b> 2009</b> recorded the highest number of female penguins.</p>

<h1>Question #12: According to the theory of evolution, species changes over time. Is there an increase of average bill length, depth, and body mass from 2007 to 2009?</h1>

<h5>Average bill length from each year</h5>

In [129]:
Blength_PerYear = penguins.groupby(['year'])['bill_length_mm'].mean()

In [130]:
Blength_PerYear 

year
2007    43.742018
2008    43.541228
2009    44.448516
Name: bill_length_mm, dtype: float64

<h5>Average bill depth from each year</h5>

In [131]:
Bdepth_PerYear =  penguins.groupby(['year'])['bill_depth_mm'].mean()

In [133]:
Bdepth_PerYear

year
2007    17.425011
2008    16.914035
2009    17.125426
Name: bill_depth_mm, dtype: float64

<h5>Average body mass from each year</h5>

In [134]:
Bmass_PerYear = penguins.groupby(['year'])['body_mass_g'].mean()

In [135]:
Bmass_PerYear

year
2007    4125.243222
2008    4266.666667
2009    4210.222953
Name: body_mass_g, dtype: float64

<h3>Insigth #12: According to the data above, we found out that:</h3>
<p><b>There's no clear indication of increase from the average bill length, bill depth, and body mass</p>

<h1>Data Aggregation</h1>

<h5>What is the number species that inhabitates each island</h5>

In [108]:
PerIsland = penguins.groupby(['sex','year']).size()

In [111]:
PerIsland = 'female'

<h5>What is the number of male Adelie on each island?</h5\>

In [76]:
AdelieMalePerIsland = adelie_df.groupby(['species','sex','island']).size()

In [77]:
AdelieMalePerIsland

species  sex     island   
Adelie   female  Biscoe       22
                 Dream        27
                 Torgersen    24
         male    Biscoe       22
                 Dream        29
                 Torgersen    28
dtype: int64

<h5>Which species has the longest bill lenght?</h5>

In [78]:
per_bill_length = penguins.groupby(['species'])['bill_length_mm'].mean()

In [79]:
per_bill_length

species
Adelie       38.825144
Chinstrap    48.833824
Gentoo       47.475983
Name: bill_length_mm, dtype: float64

<h5>What is the average of each species per island?</h5>

In [80]:
avg_species_island = penguins.groupby(['species','island'])['body_mass_g'].mean()

In [81]:
avg_species_island

species    island   
Adelie     Biscoe       3709.659091
           Dream        3688.392857
           Torgersen    3715.899123
Chinstrap  Biscoe               NaN
           Dream        3733.088235
           Torgersen            NaN
Gentoo     Biscoe       5068.965761
           Dream                NaN
           Torgersen            NaN
Name: body_mass_g, dtype: float64

<h5>Which sex is the heaviest?</h5>

In [82]:
weight_per_sex = penguins.groupby(['sex'])['body_mass_g'].mean()

In [83]:
weight_per_sex

sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h1>Testing Correlation</h1>

<h3>Pearson R Correlation</h3>
<hr>
<p>Used for testing correlation between two <strong>Numerical Values</strong></p>

In [84]:
weight_flipper_cor = penguins['body_mass_g'].corr(penguins['flipper_length_mm'])

In [85]:
weight_flipper_cor

np.float64(0.8712017673060114)